src/synthesizer/sed.py

```diff
@@ -1314,18 +1314,25 @@ def calculate_ionising_photon_production_rate(
         x = self._lam
         y = (llam * self.lam / h.to(erg / Hz) / c.to(angstrom / s)).value
 
-        # Get value of luminosity at ionisation wavelength
-        ionisation_y = np.interp(
-            ionisation_wavelength.to(angstrom).value, x, y
-        )
-
         # Restrict arrays to ionisation regime
         x = x[ionisation_mask]
-        y = y[ionisation_mask]
+        if len(y.shape) == 1:
+            y = y[ionisation_mask]
+        else:
+            y = y[:, ionisation_mask]
```

Comment:
This only works for 1D and 2D now, what about 3D?  

Use
```
masked_array = arr[..., mask]
```
to mask the final axis of an arbitrarily dimensional array.

Reply:
Fixed.